competitors/rmi_cpp_robust.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <fstream>
+
+#include "./analysis-rmi/include/rmi/models.hpp"
+#include "./analysis-rmi/include/rmi/rmi_robust.hpp"
+#include "base.h"
+
+// Alternate implementation of RMI in C++ with robust outlier compensation
+template <class KeyType, typename Layer1, typename Layer2,
+    template <typename...> typename RMIType, size_t layer2_size,
+    uint32_t variant_num>
+class RMICppRobust : public Competitor {
+ public:
+  uint64_t Build(const std::vector<KeyValue<KeyType>>& data) {
+    std::vector<KeyType> loading_data;
+    loading_data.reserve(data.size());
+    for (auto& itm : data) {
+      loading_data.push_back(itm.key);
+    }
+
+    return util::timing([&] {
+      auto rmi_ptr_one = std::make_unique<RMIType<KeyType, Layer1, Layer2>>(
```

Comment:
Let's build them one by one to save memory.

Reply:
Done